.travis.yml

```diff
@@ -0,0 +1,49 @@
+dist: trusty
+sudo: off
+
+language: node_js # Base language is Nodejs (used for liniting and basic validation).
+
+addons:
+  apt:
+    sources:
+    - sourceline: 'deb https://packages.microsoft.com/repos/azure-cli/ trusty main'
+      key_url: 'https://packages.microsoft.com/keys/microsoft.asc'
+    packages:
+      - azure-cli # Install the Azure CLI tools
+
+before_script:
+  - npm install -g grunt-cli # Install task runners for lint checking.
+
+node_js:
+  - "0.12"
+
+cache:
+  directories:
+    - node_modules
+
+script: # Run the lint tests, then if env vars are present validate the template then try building the stack.
+  - npm test
+  - if [ "$SPNAME" != ""  -a  "$SPPASSWORD" != ""  -a  "$SPTENANT" != ""  -a  "$SPSSHKEY" != "" ]; then
+      echo "Running Azure setup steps.";
+      az login --service-principal -u "$SPNAME" -p "$SPPASSWORD" --tenant "$SPTENANT";
+      az group create -l westus -g "moodle-azure-$TRAVIS_BUILD_NUMBER";
```

Comment:
This will now default to west us, unless there is a env variable of LOCATION defined.
This way it'll work as a default and allow individual override.

Reply:
Great. Thanks!